### PR TITLE
Allow death dismissal streamlining only when deceased veteran is appellant

### PR DESCRIPTION
**Pull request by nanotone**
Resolves [CASEFLOW-1676](https://vajira.max.gov/browse/CASEFLOW-1676)

### Description
This PR prevents death dismissal streamlining for deceased veterans when the appellant is not the veteran.

### Acceptance Criteria
- [ ] Code compiles correctly

### Testing Plan
1. Added new test, which fails on old code

### CODE REVIEW

File: app/workflows/initial_tasks_factory.rb [hunk 1]
```diff
@@ -156,4 +156,11 @@ def create_remand_subtask
       fail "Unsupported remand subtype: #{@cavc_remand.remand_subtype}"
     end
   end
+
+  def should_streamline_death_dismissal?
+    return false if @appeal.appellant_is_not_veteran
+    return false if @appeal.veteran.date_of_death.nil?
```
Comment (nweinmeister) [inline app/workflows/initial_tasks_factory.rb:162]: Might be worth adding an `alive?` method to the `Veteran` model. Seems like this check could be common and might make it more clear?

**Review by nweinmeister — APPROVED** [review]
Logic looks good, had one comment for styling but not super important.

File: app/workflows/initial_tasks_factory.rb (hunk 1 above)
Comment (nweinmeister) [reply app/workflows/initial_tasks_factory.rb:162]: Update: there is an `alive?` method and a `deceased?` method already, you should use those!

Comment (nanotone) [reply app/workflows/initial_tasks_factory.rb]: Great suggestion, updated.

**Pull request merged**